libs/src/services/ethContracts/serviceProviderFactoryClient.js

```diff
@@ -92,32 +96,41 @@ class ServiceProviderFactoryClient extends ContractClient {
     }
   }
 
-  async decreaseStake (amount) {
-    let method = await this.getMethod('decreaseStake', amount)
-    let tx = await this.web3Manager.sendTransaction(method, 1000000)
+  /**
+   * @param {BN} amount
+   * TODO: This method will need to broken out because the waitForBlock duration is a long time
+   * but as a proof of concept, this is how it can/should look.
+   */
+  async requestDecreaseStake (amount) {
+    const requestDecreaseMethod = await this.getMethod('requestDecreaseStake', amount)
+    await this.web3Manager.sendTransaction(
+      requestDecreaseMethod,
+      1000000
+    )
+
+    const account = this.web3Manager.getWalletAddress()
+    const lockupExpiryBlock = await this.getLockupExpiry(account)
+    return lockupExpiryBlock
+  }
+
+  async getLockupExpiry (account) {
+    const requestInfoMethod = await this.getMethod('getPendingDecreaseStakeRequest', account)
```

Comment:
might make more sense to actually expose `getPendingDecreaseStakeRequest` and have `getLockupExpiry` call into that and filter only lockupExpiry (thats already happening below)

Reply:
sg! i think this API will change a bit more as we actually get into the dashboard more